ceno_zkvm/src/scheme/prover.rs

```diff
@@ -150,6 +150,7 @@ impl<E: ExtensionField, PCS: PolynomialCommitmentScheme<E>> ZKVMProver<E, PCS> {
                 for lk_s in cs.lk_expressions_namespace_map.iter() {
                     tracing::debug!("opcode circuit {}: {}", circuit_name, lk_s);
                 }
+                transcript.append_field_element(&E::BaseField::from(num_instances as u64));
```

Comment:
Assuming it is line 131 above that does the skipping of empty circuits:

```rust
            if witness.is_empty() {
                continue;
```

Then to say "there is an empty circuit" the `num_instances` (potentially 0) should be written before that `if`.